src/signalrclient/websocket_transport.h

```diff
@@ -46,6 +46,7 @@ namespace signalr
         signalr_client_config m_signalr_client_config;
 
         std::shared_ptr<cancellation_token_source> m_receive_loop_cts;
+        std::shared_ptr<cancellation_token_source> m_receive_loop_done_cts;
```

Comment:
Maybe rename to `m_receive_loop_task` to indicate this is basically just a promise and not a way to request cancellation.